pilot/pkg/security/model/authentication.go

```diff
@@ -201,9 +204,30 @@ func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, metadata *model.N
 	}
 }
 
-// ApplyCustomSDSToCommonTLSContext applies the customized sds to CommonTlsContext
+// ApplyCustomSDSToClientCommonTLSContext applies the customized sds to CommonTlsContext
+// Used for building upstream TLS context for egress gateway's TLS/mTLS origination
+func ApplyCustomSDSToClientCommonTLSContext(tlsContext *tls.CommonTlsContext, tlsOpts *networking.ClientTLSSettings, sdsUdsPath string) {
+	// create SDS config for gateway to fetch key/cert from agent.
```

Comment:
This function needs to check tlsOpts.Mode, just like ApplyCustomSDSToServerCommonTLSContext().
If it is networking.ClientTLSSettings_SIMPLE, then only need to configure tlsContext.ValidationContextType. SIMPLE means TLS handshake, the client side only needs a server side root cert, and use that root cert to validate the server cert sent from the remote server.
The tlsContext.TlsCertificateSdsSecretConfigs() is to get the client side key/cert via SDS, which are sent to the remote server side during a mTLS handshake.

Reply:
Thanks totally missed it! Can you review if it looks good now